DatabaseDefinition.Test/Item/ItemRepositoryTests.cs

```diff
@@ -0,0 +1,123 @@
+            var testItem2 = CreateRandomItem();
+            var ids = await AddItemsAsync(testItem0, testItem1, testItem2).ConfigureAwait(false);
+            var item = await itemRepository.GetItemAsync(ids.ElementAt(0), CancellationToken).ConfigureAwait(false);
+            Assert.NotNull(item);
+            Assert.Equal(ids.ElementAt(0), item.Id);
+            Assert.True(ItemEqualsItem(testItem0, item));
+        }
+
+        [Fact]
+        public async Task GetItemReturnsNullForUnknownTest()
+        {
+
+            var testItem0 = CreateRandomItem();
+            var testItem1 = CreateRandomItem();
+            var testItem2 = CreateRandomItem();
+            _ = await AddItemsAsync(testItem0, testItem1, testItem2).ConfigureAwait(false);
+            var item = await itemRepository.GetItemAsync(50, CancellationToken).ConfigureAwait(false);  //Should throw and not just return null
+            Assert.Null(item);
+        }
+
+        [Fact]
+        public async Task UpdateItemTest()
+        {
+            var testItem0 = CreateRandomItem();
+            var testItem1 = CreateRandomItem();
+            var testItem2 = CreateRandomItem();
+            var ids = await AddItemsAsync(testItem0, testItem1, testItem2).ConfigureAwait(false);
+            var updateItem = CreateRandomItem();
+            updateItem.Id = ids.ElementAt(2);
+            await itemRepository.UpdateItemAsync(updateItem, CancellationToken).ConfigureAwait(false);
+            var item = await itemRepository.GetItemAsync(ids.ElementAt(2), CancellationToken).ConfigureAwait(false);
+            Assert.NotNull(item);
+            Assert.Equal(ids.ElementAt(2), item.Id);
+            Assert.True(ItemEqualsItem(updateItem, item));
+        }
+
+        [Fact]
+        public async Task UpdateItemThrowsForUnknownTest()
+        {
+
+            var testItem0 = CreateRandomItem();
+            var testItem1 = CreateRandomItem();
+            var testItem2 = CreateRandomItem();
+            _ = await AddItemsAsync(testItem0, testItem1, testItem2).ConfigureAwait(false);
+            var updateItem = CreateRandomItem();
+            updateItem.Id = 50;
+            await Assert.ThrowsAsync<ArgumentException>(async () =>
+                await itemRepository.UpdateItemAsync(updateItem, CancellationToken).ConfigureAwait(false)
+                ).ConfigureAwait(false);
+        }
+
+        [Fact]
+        public async Task DeleteItemTest()
+        {
+            var testItem0 = CreateRandomItem();
+            var testItem1 = CreateRandomItem();
+            var testItem2 = CreateRandomItem();
+            var ids = await AddItemsAsync(testItem0, testItem1, testItem2).ConfigureAwait(false);
+            await itemRepository.DeleteItemAsync(ids.ElementAt(1), CancellationToken).ConfigureAwait(false);
+            var item = await itemRepository.GetItemAsync(ids.ElementAt(1), CancellationToken).ConfigureAwait(false);  //Should throw and not just return null
```

Comment:
@schlechtOptimiziert  "Should throw and not just return null", wirst du das in deinem Branch ändern oder soll ich das hier machen?

Reply:
Ja